### ISSUE TITLE
Update categories for patterns to use Site Building and Universal categories

### ISSUE THREAD

**Issue by kjellr** [WordPress/pattern-directory#190]
Create new categories matching those in Gutenberg

* **Call to Action** 
Sections whose purpose is to trigger a specific action.
* **Team** 
A variety of designs to display your team members.
* **Testimonials** 
Share reviews and feedback about your brand/business.
* **Services** 
Briefly describe what your business does and how you can help.
* **Contact** 
Display your contact information.
* **About**
 Introduce yourself.
* **Portfolio**
 Showcase your latest work.
* **Media** 
Different layouts containing video or audio.
* **Posts**
 Display your latest posts in lists, grids or other layouts.
* **Products**
 Display your store’s products in lists, grids or other layouts.
* **Post Content**
 Your post and page content.
* **Comments**
Different ways of displaying your post or page's comments.
* **Pagination**
 A variety of designs for navigating your posts.
* **Comment Pagination** 
 A variety of designs to browse through a big list of comments.
* **Archive Headings**
 A variety of designs for your archive heading.
* **404**
 A variety of designs for when a page cannot be found.
* **Search**
 Different layouts to display search results.

## To do

- [ ] Create the above categories, don't remove existing categories
- [ ] Update the categories which are allowed in the pattern creator
- [ ] Update the filter menu with new categories, with toggle for site-building/content which changes the category list (separate issue, waiting on a design #395)

---------

_Original issue:_

Now that we have many more patterns in the directory, let's consider adding some additional categories. A couple early suggestions based on the patterns we're seeing so far: 

- Quote
- Image & Text

Let's discuss these and others, and we can add the ones that make sense.

**Comment by shaunandrews** [comment]
Along with adding some new categories, I think we could combine Images and Gallery into "Images."

--

I think Header is problematic; There could be confusion around the header Template Part, but also the patterns currently in the Header category aren't really _headers_. Instead of Header (which I think we should reserve for Template Part patterns) I think maybe "Cover," "Hero," or "Introduction" could be better group names. "Image & Text" could fall into this category, too.

--

We should consider adding some categories (and patterns) that focus on some actual use-cases, like "Blogging," "Portfolio," or "Events."

--

- Blogging
- Buttons
- Columns
- Cover
- Events
- Images
- Portfolio
- Text

**Comment by richtabor** [comment]
Or "Headlines" perhaps, instead of a "Header" or "Hero" category 🤔

**Comment by shaunandrews** [comment]
Headlines works well.

**Comment by jasmussen** [comment]
https://github.com/WordPress/gutenberg/issues/44501 suggests a big revamp of default block editor categorization, and includes a suggestion for the pattern directory as well

* New categories for post/page patterns
* New categories for site building patterns
* A way to site building patterns only in the editor
* A splitting of the pattern directory into "Featured" and "Community" patterns to enable https://github.com/WordPress/gutenberg/issues/44611 

I see the "Headlines" conversation was missed in the core conversation, no strong opinions there but would be good to sync up so categories can be the same. 

What would be a best next step to move this forward for the pattern directory?

**Comment by ryelle** [comment]
@jasmussen The list on https://github.com/WordPress/gutenberg/issues/44501 is a good start.

> New categories for post/page patterns, site building patterns

I can add these new categories to the pattern creator. Should the old categories be removed immediately? Should any current patterns be updated (or can any updates happen by a designer manually)?

Should the navigation on the pattern directory site change?

> A way to site building patterns only in the editor

I'm not totally sure what this means.

> A splitting of the pattern directory into "Featured" and "Community" patterns to enable https://github.com/WordPress/gutenberg/issues/44611

This is not a problem technically, we can use the internal keyword system. "Featured" could be the tag, and everything else is "Community" (unless there might be patterns with neither categorization?)

 I think there's a larger conversation around how this will be used, though. How (UX-wise) should patterns be added to the "featured" term? Who will do that moderation? Will there be public criteria for what is "featured"?

Will this "featured" split be visible on the pattern directory site anywhere? There's [this idea of a toggle control in the pattern explorer](https://github.com/WordPress/gutenberg/issues/44611#issuecomment-1291768662), we could do something similar in the filter dropdown? Switch between featured and all, or featured/community/all…?

**Comment by jasmussen** [comment]
> Should the old categories be removed immediately? Should any current patterns be updated (or can any updates happen by a designer manually)?

I like to think that so much has happened in 6.1 with content locking, intrinsic responsive behaviors, that there's a good opportunity for a fresh start, but I'm not sure how to best handle that, either technically or in terms of the submissions already there, if there was an "archiving" functionality where a submitter can go and update and recategorize their pattern. Alternatively we can probably establish some migration policies mapping old categories to the new, if that would be helpful?

> Should the navigation on the pattern directory site change?

Probably yes, I'll see if we can't get some designs soon, both near term and as part of the WP.org refresh.

> A way to site building patterns only in the editor

My apologies, I got that way wrong, must've been a monday. I meant to say, "A way to show site building patterns only in the site editor".

> I think there's a larger conversation around how this will be used, though. How (UX-wise) should patterns be added to the "featured" term? Who will do that moderation? Will there be public criteria for what is "featured"?
> Will this "featured" split be visible on the pattern directory site anywhere? There's https://github.com/WordPress/gutenberg/issues/44611#issuecomment-1291768662, we could do something similar in the filter dropdown? Switch between featured and all, or featured/community/all…?

Yes, the featured split should default to showing the featured patterns. It could be a toggle, or it could be two tabs. The second level would show the row of categories, just like they exist today. The only patterns that go in the Featured tab are those from the WordPress.org user. These patterns would be equivalent to Twenty themes, so not something that gets updated all that often. Essentially I picture these patterns as being the same as the default patterns that are bundled in the main block editor inserter.

**Comment by ryelle** [comment]
>  if there was an "archiving" functionality where a submitter can go and update and recategorize their pattern.

Patterns can be edited by the author (although it does push them back through the review queue in case the content's changed). Patterns can also be set to draft or trashed if they're really irrelevant.

> Alternatively we can probably establish some migration policies mapping old categories to the new, if that would be helpful?

Yes, I think this will be necessary. Can you provide this?

> A way to show site building patterns only in the site editor

I don't think we need to worry about that in the pattern directory, just in Gutenberg (so it's not relevant to this issue).

> The only patterns that go in the Featured tab are those from the WordPress.org user. 

Okay, we can start with that, and figure out the moderation strategy after some wider discussion.

**Comment by jasmussen** [comment]
Here's a stab, would appreciate a sanity check, CC: @beafialho @jameskoster

* Featured remains in the editor, as a curated category. Not on the directory, there I guess it's "all" that's the closest counterpart.
* Buttons → Call to Action
* Columns → Media
* Footers → Footers
* Gallery → Gallery
* Headers → Headers
* Text → Media
* Posts → Posts
* Wireframe → Wireframe

The main odd ones are "Columns" and "Text", and in many ways feel like the motivation for the newer and less generic categories (CTA, Testimonials, Team, Contact, About, Portfolio, Products) — so ideally they could be sorted in those categories. But as a 1:1 transfer, "Media" IMO feels like the best kitchensink for it. What do you think?

**Comment by jameskoster** [comment]
Hmm, yeah that's not a great fit, especially as the Media description specifically mentions video and audio. It would be very strange to find testimonial patterns in the Media category when a dedicated Testimonials category exists. 

In an ideal world those patterns would be re-classified on a case by case basis. Basically all of the ones in the [Text](https://wordpress.org/patterns/categories/text/) category should be migrated to the new Call to Action / About / Testimonials / Team / Services categories accordingly. Likewise for Columns. I noticed _many_ of the patterns in these categories include keywords in their names, could we write a script to auto-sort them? 🤔 

I do wonder whether we might also include a category for generic content compositions like [this one](https://wordpress.org/patterns/pattern/two-columns-of-text/). "Text layouts" or something?

**Comment by richtabor** [comment]
> Basically all of the ones in the [Text](https://wordpress.org/patterns/categories/text/) category should be migrated to the new Call to Action / About / Testimonials / Team / Services categories accordingly.

Agreed. 

A catch-all may be necessary, as @jameskoster mentioned, but I do think that having more specific patterns in these content-oriented categories is the way to go. 

When looking for a pattern, is the thought "I want to add some text here" or is it "I want to add a services section here". If we don't re-organize, then the second isn't possible to see/browse (although you can search fine).

**Comment by ryelle** [comment]
@jasmussen This is a helpful start, but I think we actually need the reverse — some mapping from the new to the old.

For the current purposes, I'm trying to accomplish two things with the mapping question:

1. The new category pages on the pattern directory (ex, `/patterns/categories/call-to-action/`), shouldn't all be empty once the new nav is implemented. So by adding in some mapping, we can show the current patterns in that space (ex, patterns tagged buttons can show up under Call to Action)
    - As patterns are created with the new cats, this will be less of a problem, and we can eventually remove that.
2. When returning the pattern list to a WP site that doesn't support the new categories (a site running WP 6.1 without GB), they shouldn't all be dumped into "uncategorized", so a pattern tagged "call to action" can show up in the "buttons" category on that WP site.

These could be two separate lists, or maybe one or both is deemed unnecessary. What do you think?

**Comment by ryelle** [comment]
I just realized that https://github.com/WordPress/gutenberg/issues/44501 is still open, and that https://github.com/WordPress/gutenberg/pull/45548/ is a draft, so technically no one has these categories yet. I'm going to continue breaking down the tasks here since I've already started, but I don't think we should be changing the pattern directory until these categories are in gutenberg.

**Comment by jasmussen** [comment]
> When returning the pattern list to a WP site that doesn't support the new categories (a site running WP 6.1 without GB), they shouldn't all be dumped into "uncategorized", so a pattern tagged "call to action" can show up in the "buttons" category on that WP site.

Can you expand that a bit for me? It was my impression that there wasn't _yet_ any connection between WordPress and the pattern directory.

**Comment by ryelle** [comment]
Currently, `core` and `featured` patterns are auto-loaded into WordPress sites, and theme authors can set specific patterns to be loaded using pattern slugs in the `theme.json` file. If the categories changed right now, 3 of these remote-loaded patterns would not have registered categories on the site ([1](https://wordpress.org/patterns/pattern/two-columns-of-text-with-offset-heading/), [2](https://wordpress.org/patterns/pattern/quote/), [3](https://wordpress.org/patterns/pattern/two-columns-of-text/)), plus any theme authors might be loading.

It's not a large number (which is why I said it might be unnecessary to handle it) - TBH I thought it was more like 8 when I first flagged it, but since Gallery and Headers are still available it's just those 3.

**Comment by annezazu** [comment]
@ryelle with https://github.com/WordPress/gutenberg/pull/46144 merged to bring the new categories, can this work now move forward? Wanting to check in to see what else might be blocking these changes!

**Comment by ryelle** [comment]
The categories were added temporarily, but the way the site works, the new cats accidentally took over the filter navigation. Right now, we're waiting on a design to be finished for the filters (which can be added to #395), so that when the categories are added, they'll just slot right into the new design.

**Comment by ryelle** [comment]
I've added the new categories to the site on the backend, and prevented them from showing up on the frontend until we have a design (#395). This way they can be used in the API responses in Gutenberg. They're also already available for in the pattern creator, so when we launch the new design there should be a few patterns with the new categories.

I'm going to close this issue because I think the rest of this will be covered in #395 & #542. If there are any problems with these new categories in the editor, we can create new issues.
